spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+  describe '#index' do
+    let!(:user) { FactoryBot.create(:user) }
+
+    context "when user is logged in" do
+      context "and gives no search params" do
+        before(:each) do
+          sign_in user
+          get :index
+        end
+
+        it 'responds with a status code of 200' do
+          expect(response.status).to eq 200
+        end
+
+        it 'renders the index page' do
+          expect(response).to render_template(:index)
+        end
+
+        it 'assigns users to all users' do
+          expect(assigns[:users]).to eq User.all
+        end
+      end
+
+      context "and gives search params" do
+        before(:each) do
+          sign_in user
+          get :index, params: { search: user.email.downcase }
```

Comment:
This isn't a problem with your test, but it's probably worth downcasing all user info (name, email, etc.) as it comes into the app. We can always call .upcase on it when we render it later.